src/GameKit/GameKit.cs

```diff
@@ -147,31 +190,60 @@ public enum GKError : long {
 		FriendRequestNotAvailable = 103,
 	}
 
+#if NET
+	[SupportedOSPlatform ("ios10.0")]
+	[SupportedOSPlatform ("macos10.12")]
+	[SupportedOSPlatform ("tvos10.0")]
+#else
+	[iOS (10,0)]
+	[Mac (10,12)]
+	[TV (10,0)]
+#endif
 	[Native]
-	[iOS (10,0)][Mac (10,12)][TV (10,0)]
 	public enum GKConnectionState : long {
 		NotConnected,
 		Connected,
 	}
 
+#if NET
+	[SupportedOSPlatform ("ios10.0")]
+	[SupportedOSPlatform ("macos10.12")]
+	[SupportedOSPlatform ("tvos10.0")]
+#else
+	[iOS (10,0)]
+	[Mac (10,12)]
+	[TV (10,0)]
+#endif
 	[Native]
-	[iOS (10,0)][Mac (10,12)][TV (10,0)]
 	public enum GKTransportType : long {
 		Unreliable,
 		Reliable,
 	}
 
+#if NET
+	[UnsupportedOSPlatform ("macos10.14")]
+	[UnsupportedOSPlatform ("tvos12.0")]
+	[UnsupportedOSPlatform ("ios12.0")]
+#if MONOMAC
+	[Obsolete ("Starting with macos10.14.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#elif TVOS
+	[Obsolete ("Starting with tvos12.0.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#elif IOS
+	[Obsolete ("Starting with ios12.0.", DiagnosticId = "BI1234", UrlFormat = "https://github.com/xamarin/xamarin-macios/wiki/Obsolete")]
+#endif
+#else
 	[Deprecated (PlatformName.MacOSX, 10,14)]
 	[Deprecated (PlatformName.TvOS, 12,0)]
 	[Deprecated (PlatformName.iOS, 12,0)]
 	[Native]
-#if WATCH && !XAMCORE_4_0
+#if WATCH
 	// removed in Xcode 10 but a breaking change (for us) to remove
 	[Obsolete ("Not used in watchOS.")]
 #else
 	[Unavailable (PlatformName.WatchOS)]
 	[ErrorDomain ("GKGameSessionErrorDomain")]
 #endif
+#endif // NET
```

Comment:
Could I get extra eyes on this change above?
I was having trouble with GKGameSessionErrorCode not being exposed when I moved things around 
after the `#if WATCH`

Is any of this unnecessary?

Reply:
LGTM

Reply:
same.